DESIGN.md

```diff
@@ -0,0 +1,818 @@
+
+The `id` field of a `<measurement-entry>` message is key to get the
+raw measurement JSON, using `ooniprobe --batch show <id>`.
+
+#### Content of the test_keys field
+
+For the `"websites"` experiment group:
+
+```JSON
+{
+  "accessible": true,
+  "blocking": "none"
+}
+```
+
+where `accessible` indicates whether the URL is accessible and `blocking`
+indicates they way in which we believe the URL to be blocked.
+
+For the `"dash"` experiment in the `"performance"` group:
+
+```JSON
+{
+  "connect_latency": 0.06374096870422363,
+  "median_bitrate": 36724, 
+  "min_playout_delay": 0
+} 
+```
+
+where `"connect_latency"` is the time to establish a connection,
+`"median_bitrate"` is the median bitrate in kbit/s, and
+`"min_playout_delay`" is the minimum delay required to perform
+a smooth streaming, given the streaming speed.
+
+For the `"ndt"` experiment in the `"performance"` group:
+
+```JSON
+{
+  "upload": 21125.226149733466,
+  "download": 76153.12071218148,
+  "ping": 64
+}
+```
+
+where `"upload"` is the upload speed in kbit/s, `"download"` is the
+download speed in kbit/s, and `"ping"` is the minimum RTT in ms.
+
+For the experiments in the `"middlebox"` group:
+
+```JSON
+{}
+```
+
+For the `"facebook_messenger"` experiment in the `"im"` group:
+
+```JSON
+{
+  "facebook_dns_blocking": false,
+  "facebook_tcp_blocking": false
+}
+```
+
+where `"facebook_dns_blocking"` indicates whether Facebook Messenger
+is blocked at the DNS level, and `"facebook_tcp_blocking"` indicates
+whether it is blocked at the TCP/IP level.
+
+For the `"telegram"` experiment in the `"im"` group:
+
+```JSON
+{
+  "telegram_http_blocking": false,
+  "telegram_tcp_blocking": false,
+  "telegram_web_blocking": false,
+}
+```
+
+where the keys have the following meaning:
+
+- `"telegram_http_blocking"` indicates whether the connection to some
+selected telegram servers is blocked by HTTP blocking, resulting in the
+telegram app not being able to work
+
+- `"telegram_tcp_blocking"` indicates whether the connection to some
+selected telegram servers is blocked by TCP blocking, resulting in the
+telegram app not being able to work
+
+- `"telegram_web_blocking"` indicates instead whether the telegram web
+interface is blocked
+
+For the `"whatsapp"` experiment in the `"im"` group:
+
+```JSON
+{
+  "registration_server_blocking": false,
+  "whatsapp_web_blocking": false,
+  "whatsapp_endpoints_blocking": false
+} 
+```
+
+where:
+
+- `"registration_server_blocking"` indicates whether the registration
+server is blocked
+
+- `"whatsapp_web_blocking"` indicates whether the web interface
+to WhatsApp is blocked
+
+- `"whatsapp_endpoints_blocking"` indciates whether selected endpoints
+used by the WhatsApp app are blocked
+
+### Getting a measurement JSON
+
+This command:
+
+```
+ooniprobe --batch show <id>
+```
+
+emits in output the measurement identified by `<id>`. The data
+format is described at https://github.com/ooni/spec.
```

Comment:
It would be cool if this was done by fetching it from the API like we do on mobile, though this is maybe something post-MVP.

Reply:
Yes. Also, this is an implementation detail. As far as the probe-desktop is concerned, this is the command to show a measurement and it doesn't matter how it's done.